src/ExpEvaluator.cpp

```diff
@@ -219,8 +216,8 @@ FP_PRECISION ExpEvaluator::computeExponential(FP_PRECISION tau, int polar) {
 
   /* Evaluate the exponential using the lookup table - linear interpolation */
   if (_interpolate) {
-    int index;
-    index = floor(tau * _inverse_exp_table_spacing);
+    tau = std::min(tau, (_max_optical_length));
+    int index = floor(tau * _inverse_exp_table_spacing);
```

Comment:
Let's assume tau is set to the maximum possible value on line 219, `_max_optical_length`. On lines 171-175, we see that the formula for the `_inverse_exp_table_spacing` is:

`_inverse_exp_table_spacing = 1.0 / _exp_table_spacing = num_array_values / _max_optical_length`

This would result in line 220 being equivalent to:

`int index = floor(tau * _inverse_exp_table_spacing) = floor(_max_optical_length * num_array_values / _max_optical_length) = floor(num_array_values) = num_array_values`

In short, if `tau` is set to `_max_optical_length`, the index will be one greater than the max index of the array. I confirmed this by finding the max optical path length of any segment for the `one-directional.py` input and setting `_max_optical_length` to this value. 

How about incrementing `num_array_values` in `ExpEvaluator::initialize()` after line 172, but before line 181? This would increase the size of the table by one to account for cases where `tau` is equal to `_max_optical_length`.